### ISSUE TITLE
conda update --all fails

### ISSUE THREAD

**Issue by gvdeynde** [conda/conda#3529]
InvalidTypeError: Parameter proxy_servers = 'None' declared in <<merged>> has type <class 'str'>.
Valid types:   - <class 'dict'>.

Current conda install:

```
           platform : win-64
      conda version : 4.2.8
   conda is private : False
  conda-env version : 4.2.8
conda-build version : 2.0.1
     python version : 3.5.2.final.0
   requests version : 2.11.1
   root environment : C:\Anaconda3  (writable)
default environment : C:\Anaconda3
   envs directories : C:\Anaconda3\envs
      package cache : C:\Anaconda3\pkgs
       channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                      https://repo.continuum.io/pkgs/free/noarch/
                      https://repo.continuum.io/pkgs/pro/win-64/
                      https://repo.continuum.io/pkgs/pro/noarch/
                      https://repo.continuum.io/pkgs/msys2/win-64/
                      https://repo.continuum.io/pkgs/msys2/noarch/
        config file : c:\users\gvdeynde\.condarc
       offline mode : False
```

**Comment by kalefranz** [comment]
What is the output of `conda config --show-sources`?

**Comment by kalefranz** [comment]
Also the contents of `c:\users\gvdeynde\.condarc`?

**Comment by gvdeynde** [comment]
output of conda config --show-sources is same as of conda update --all:

```
InvalidTypeError: Parameter proxy_servers = 'None' declared in <<merged>> has type <class 'str'>.
Valid types:   - <class 'dict'>.
```

contents of .condarc:

```
proxy_servers:
#    http: http://192.168.3.1:8080
#    https: https://192.168.3.1:8080

ssl_verify: false
channels:
  - defaults
```

And that is the culprit: conda fails because the line proxy_servers is not commented while the proxies themselves are commented out. Don't know if this is error message is desired behaviour though.

**Comment by kalefranz** [comment]
It's desired behavior actually. The real missing piece is better messaging about *why*, and better hints on what you need to do to fix it :)

**Comment by mrhupp** [comment]
I found a related error not fixed by instructions above.

The contents of my system .condarc with no spaces before the http: and https: entries:
```
proxy_servers:
http: http://web-proxy.blah.blah.com:8080/
https: https://web-proxy.blah.blah.com:8080/
ssl_verify: False
```
This worked fine for months until today when a conda update --all resulted in:
```
'InvalidTypeError: Parameter proxy_servers = u'None' declared in <<merged>> has type <type 'unicode'>.
Valid types:   - <type 'dict'>.'
```
After seeing the fix above and then looking at [the proxy_server section of the Conda Configuration document](http://conda.pydata.org/docs/config.html#configure-conda-for-use-behind-a-proxy-server-proxy-servers), I tried putting a couple spaces at the front of the http: and https: entries like this:
```
proxy_servers:
  http: http://web-proxy.blah.blah.com:8080/
  https: https://web-proxy.blah.blah.com:8080/
ssl_verify: False
```
and it started working again.

**NOTE:** Nowhere in [the proxy_server section of the Conda Configuration document](http://conda.pydata.org/docs/config.html#configure-conda-for-use-behind-a-proxy-server-proxy-servers) does it say the spaces are required and they weren't until a short time ago (which I cannot pinpoint).

**Comment by kalefranz** [comment]
Stack trace recently fixed in 4.3.17.
